src/api/spec/controllers/webui/apidocs_controller_spec.rb

```diff
@@ -2,26 +2,40 @@
 
 RSpec.describe Webui::ApidocsController, type: :controller do
   describe "GET #index" do
+    let!(:previous_apidocs_location) { CONFIG['apidocs_location'] }
+
+    after do
+      CONFIG['apidocs_location'] = previous_apidocs_location
+    end
+
     context "correct setup" do
+      let(:tmp_dir) { Dir.mktmpdir }
+      let(:tmp_file) { "#{tmp_dir}/index.html" }
+
       before do
+        File.open(tmp_file, 'w') do |f|
+          f.write('<html><head></head><body></body></html>')
+        end
+        CONFIG['apidocs_location'] = tmp_dir
+
         get :index
       end
 
+      after do
+        File.delete(tmp_file)
+        Dir.rmdir(tmp_dir)
+      end
+
       it "responses without error" do
         expect(response).to have_http_status(:ok)
         expect(response.content_type).to eq "text/html"
       end
     end
 
     context "broken setup" do
-      let!(:old_location) { CONFIG['apidocs_location'] }
-
       before do
-        CONFIG['apidocs_location'] = '/your/mom'
-      end
-
-      after do
-        CONFIG['apidocs_location'] = old_location
+        CONFIG['apidocs_location'] = 'non/existent/subdirectory'
+        allow(File).to receive(:exist?).and_return(false)
```

Comment:
Is it necessary to mock File.exists? I'd prefer if it wasn't mocked so as not to make test too specific to the implementation of the controller.